### PR TITLE
Convert/java to kotlin

### LINKED ISSUE #2: Convert Java code to Kotlin
* Follow the procedure from the Android Docs `Java to Kotlin` conversion guide to get this done.
* Props for code enhancements and refactoring.

### PR DESCRIPTION
**Pull request by gracenudze**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -46,4 +47,8 @@ dependencies {
     implementation 'com.github.hadibtf:SemiCircleArcProgressBar:1.1.1'
     implementation 'com.github.lzyzsd:CircleProgress:v1.2.4'
     implementation 'com.github.Shinelw:ColorArcProgressBar:ebcaff93c8'
+    implementation "androidx.core:core-ktx:1.7.0"
 }
+//repositories {
+//    mavenCentral()
+//}
```
Comment (James-Aidoo) [inline app/build.gradle]: remove commented code

File: app/src/main/java/com/example/moneymanager/ActivityList.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.example.moneymanager
+
+class ActivityList(val activityIcons: Int, val activityNames: String, val activityDates: String, val balance: String)
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/ActivityList.kt]: There doesn't seem to be anything "Listy" about this class. Consider renaming it. Also, you should probably make it a data class

It is considered good practice to leave an empty line at the end of your files

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import java.util.ArrayList
+
+class GamingRoomActivity : AppCompatActivity() {
+    var activityList = ArrayList<ActivityList>()
+    var activityLogos = intArrayOf(R.drawable.ic_shopping_cart, R.drawable.withdrawal_icon,
+            R.drawable.gaming_icon)
+    var activity_names = arrayOf("Shopping", "Withdrawal", "Gaming")
+    var activity_dates = arrayOf("Jan 24, 2022", "Jan 23, 2022", "Jan 21, 2022")
+    var activity_balance = arrayOf("-$54", "-$54", "-$54")
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt]: Follow naming convention: `activity_names` > `activityNames`, `activity_dates` > `activityDates`, `activity_balance` > `activityBalance`.

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.ImageButton
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.RecyclerView
+import java.util.ArrayList
+
+class GamingRoomActivity : AppCompatActivity() {
+    var activityList = ArrayList<ActivityList>()
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt]: For all your mutable (`var`) variables, if you do not reassign new values to them anywhere later in your code, make them immutable (`val`).

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    private fun createActivityList() {
+        var count = 0
+        for (names in activity_names) {
+            val activityList = ActivityList(activityLogos[count], names,
+                    activity_dates[count], activity_balance[count])
+            this.activityList.add(activityList)
+            count++
+        }
+    }
+}
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt]: It is considered good practice to leave an empty line at the end of your files

File: app/src/main/java/com/example/moneymanager/GamingRoomAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        holder.activityNames.text = activityList.activityNames
+        holder.activityDates.text = activityList.activityDates
+        holder.balance.text = activityList.balance
+    }
+
+    override fun getItemCount(): Int {
+        return mActivityList.size
+    }
+}
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/GamingRoomAdapter.kt]: It is considered good practice to leave an empty line at the end of your files

File: app/src/main/java/com/example/moneymanager/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import android.widget.Toast
+import java.util.ArrayList
+
+class MainActivity : AppCompatActivity() {
+    var transactionList = ArrayList<TransactionList>()
+    var transactionLogos = intArrayOf(R.drawable.bag, R.drawable.ic_cloud_queue)
+    var transactionNames = arrayOf("Groceries", "Monthly iCloud")
+    var transactionDates = arrayOf("Jan 24, 2022", "Jan 22, 2022")
+    var budget = arrayOf("-$54", "-$29")
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/MainActivity.kt:20]: For all your mutable (`var`) variables, if you do not reassign new values to them anywhere later in your code, make them immutable (`val`).

File: app/src/main/java/com/example/moneymanager/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        createTransactionList()
+        val transactionRecycler = findViewById<RecyclerView>(R.id.rvTransaction)
+        transactionRecycler.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL,
+                false)
+        val firstSection = TransactionAdapter(transactionList)
+        transactionRecycler.adapter = firstSection
+    }
+
+    private fun createTransactionList() {
+        for ((count, names) in transactionNames.withIndex()) {
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/MainActivity.kt]: `count` variable does not reflect what that value actually is. Consider using a more appropriate name like `index`.

File: app/src/main/java/com/example/moneymanager/MainActivity.kt [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        bottomSheetDialog.setCanceledOnTouchOutside(false)
+        val btnCreate = bottomSheetDialog.findViewById<Button>(R.id.btnCreate)
+        btnCreate!!.setOnClickListener {
+            Toast.makeText(applicationContext, "Create Button is Clicked ",
+                    Toast.LENGTH_LONG).show()
+            bottomSheetDialog.dismiss()
+        }
+        bottomSheetDialog.show()
+    }
+}
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/MainActivity.kt]: It is considered good practice to leave an empty line at the end of your files

File: app/src/main/java/com/example/moneymanager/TransactionAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        holder.transactionNames.text = transactionList.transactionNames
+        holder.transactionDates.text = transactionList.transactionDate
+        holder.transactionBalance.text = transactionList.budget
+    }
+
+    override fun getItemCount(): Int {
+        return mTransactionList.size
+    }
+}
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/TransactionAdapter.kt]: It is considered good practice to leave an empty line at the end of your files

File: app/src/main/java/com/example/moneymanager/TransactionList.kt [hunk 11]
```diff
@@ -0,0 +1,3 @@
+package com.example.moneymanager
+
+class TransactionList(val transactionLogos: Int, val transactionNames: String, val transactionDate: String, val budget: String)
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/TransactionList.kt]: There doesn't seem to be anything "Listy" about this class. Consider renaming it. Also, you should probably make it a data class.

It is considered good practice to leave an empty line at the end of your files

File: build.gradle [hunk 12]
```diff
@@ -1,13 +1,15 @@
...
     repositories {
         google()
         mavenCentral()
             maven { url 'https://jitpack.io' }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.4'
-        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
+        classpath 'com.android.tools.build:gradle:7.1.2'
+//        classpath 'com.github.dcendents:android-maven-gradle-plugin:1.5'
```
Comment (James-Aidoo) [inline build.gradle]: Remote commented code

File: build.gradle [hunk 13]
```diff
@@ -17,4 +19,8 @@ buildscript {
 
 task clean(type: Delete) {
     delete rootProject.buildDir
-}
\ No newline at end of file
+}
+
+//repositories {
```
Comment (James-Aidoo) [inline build.gradle]: Remove commented code

It is considered good practice to leave an empty line at the end of your files

**Review by James-Aidoo — CHANGES_REQUESTED** [review]
Good job. 

However, Implement these suggestions.

File: app/src/main/java/com/example/moneymanager/ActivityList.kt (hunk 2 above)
Comment (gracenudze) [reply app/src/main/java/com/example/moneymanager/ActivityList.kt]: I named it list because it is part of the recyclerview list process

Comment (James-Aidoo) [reply app/src/main/java/com/example/moneymanager/ActivityList.kt]: I get that. However, the class isn't a list. It having `list` in it's name conveys a different message than what it's purpose actually is.

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt (hunk 4 above)
Comment (gracenudze) [reply app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt]: okay

File: app/src/main/java/com/example/moneymanager/ActivityRecycler.kt [hunk 14]
```diff
@@ -0,0 +1,3 @@
+package com.example.moneymanager
+
+data class ActivityRecycler(val activityIcons: Int, val activityNames: String, val activityDates: String, val balance: String)
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/ActivityRecycler.kt]: A more appropriate name would be `ActivityItem`, considering this class isn't  a `Recycler`View.

File: app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt [hunk 15]
```diff
@@ -31,12 +31,10 @@ class GamingRoomActivity : AppCompatActivity() {
     }
 
     private fun createActivityList() {
-        var count = 0
-        for (names in activity_names) {
-            val activityList = ActivityList(activityLogos[count], names,
-                    activity_dates[count], activity_balance[count])
+        for ((count, names) in activityNames.withIndex()) {
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/GamingRoomActivity.kt]: `index` would be more appropriate than `count`, don't you think?

File: app/src/main/java/com/example/moneymanager/TransactionRecycler.kt [hunk 16]
```diff
@@ -0,0 +1,4 @@
+package com.example.moneymanager
+
+data class TransactionRecycler(val transactionLogos: Int, val transactionNames: String, val transactionDate: String, val budget: String)
```
Comment (James-Aidoo) [inline app/src/main/java/com/example/moneymanager/TransactionRecycler.kt]: `TransactionItem`

**Review by James-Aidoo — CHANGES_REQUESTED** [review]

**Review by James-Aidoo — APPROVED** [review]

**Pull request merged**